dashboard/src/actions/config.ts

```diff
@@ -35,18 +32,28 @@ export function getConfig(): ThunkAction<Promise<void>, IStoreState, null, Confi
   };
 }
 
+// getTheme retrieves the current server configuration, calculates the proper theme
+// and stores it in both the redux state and the user's localstorage
```

Comment:
I don't understand why `getTheme` is additionally storing the gotten theme in the local storage. This is exactly what I thought the problem was: it should only ever be set in local storage iff the user explicitly makes a decision, I think? (given that it is treated as the user's preference)

For example, suppose kubeapps is installed and used by a user for the first time, and their window prefs say light. Now when this `getTheme` action is run, it determines the light theme from the browser (given no other indicators) and stores the theme in local storage. At that point, it's no longer possible to determine whether that was a user-preference or browser or system configuration. So if the installation is updated with a specific theme in the values, it won't override the previous preference of the users browser since the code determines that the light theme recorded in the local storage is now a user preference.

Everything else about this PR looks great (and clear!), just wondering if we need to store the it in local storage here, or can instead only do so when the user changes the theme explicitly.

Reply:
Yeah! You are right, we can definitely do that. I've added some changes aimed at that: I've split  the `setTheme` and `setUserTheme`. The former updates the redux state but the latter also save it in the local storage.

This other video should showcase what you meant in this comment:
https://user-images.githubusercontent.com/11535726/127335777-3e7cdef0-75e9-4af5-8fb4-1b716f13ea30.mp4

Reply:
Great thanks.